src/tests/serde.rs

```diff
@@ -755,76 +753,110 @@ fn test_datetime_helpers() {
 
     let _guard = LOCK.run_concurrently();
 
-    #[serde_as]
-    #[derive(Deserialize, Serialize)]
-    struct A {
-        #[serde_as(as = "BsonDatetimeAsRfc3339String")]
-        pub date: DateTime,
+    #[cfg(feature = "serde_with-3")]
+    {
+        #[serde_as]
+        #[derive(Deserialize, Serialize)]
+        struct A {
+            #[serde_as(as = "BsonDateTimeAsRfc3339String")]
+            pub date: DateTime,
 
-        #[serde(skip_serializing_if = "Option::is_none")]
-        #[serde_as(as = "Option<BsonDatetimeAsRfc3339String>")]
-        pub date_optional: Option<DateTime>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            #[serde_as(as = "Option<BsonDateTimeAsRfc3339String>")]
+            pub date_optional_empty: Option<DateTime>,
```

Comment:
We can skip this test case and just test `date_optional_none` and `date_optional_some`. The `#[serde(skip_serializing_if = ...)]` functionality is not actually implemented by us, so we're not getting much useful test coverage of our own code by testing both scenarios.